pkg/validate/image.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kubernetes-incubator/cri-tools/pkg/framework"
+	internalapi "k8s.io/kubernetes/pkg/kubelet/api"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	// image name for test image api
+	testImageName = "busybox"
+
+	// name-tagged reference for test image
+	testImageRef = testImageName + ":1.26.2"
+
+	// Digested reference for test image
+	busyboxDigestRef = testImageName + "@sha256:817a12c32a39bbe394944ba49de563e085f1d3c5266eb8e9723256bc4448680e"
+)
+
+var _ = framework.KubeDescribe("Image Manager", func() {
+	f := framework.NewDefaultCRIFramework()
+
+	var c internalapi.ImageManagerService
+
+	BeforeEach(func() {
+		c = f.CRIClient.CRIImageClient
+	})
+
+	It("public image with tag should be pulled and removed [Conformance]", func() {
+		testPullPublicImage(c, testImageRef)
+	})
+
+	It("public image without tag should be pulled and removed [Conformance]", func() {
+		testPullPublicImage(c, testImageName)
+	})
+
+	It("public image with digest should be pulled and removed [Conformance]", func() {
+		testPullPublicImage(c, busyboxDigestRef)
+	})
+
+	It("image status get image fields should not be empty [Conformance]", func() {
+		pullPublicImage(c, testImageRef)
+
+		defer removeImage(c, testImageRef)
+
+		status := imageStatus(c, testImageRef)
+		Expect(status.Id).NotTo(BeNil(), "image Id should not be nil")
+		Expect(len(status.RepoTags)).NotTo(Equal(0), "should have repoTags in image")
+		Expect(status.Size_).NotTo(BeNil(), "image Size should not be nil")
+	})
+
+	It("ListImage should get exactly 3 image in the result list [Conformance]", func() {
+		// different tags refer to different images
+		testImageList := []string{
+			"busybox:1-uclibc",
+			"busybox:1-musl",
+			"busybox:1-glibc",
+		}
+
+		pullImageList(c, testImageList)
+
+		defer removeImageList(c, testImageList)
+
+		images := listImages(c, testImageName)
```

Comment:
I think list and filter out dedicated is ok (only one API call instead of three).